cmd/tailscale/cli/up.go

```diff
@@ -112,6 +112,7 @@ func newUpFlagSet(goos string, upArgs *upArgsT, cmd string) *flag.FlagSet {
 	upf.StringVar(&upArgs.hostname, "hostname", "", "hostname to use instead of the one provided by the OS")
 	upf.StringVar(&upArgs.advertiseRoutes, "advertise-routes", "", "routes to advertise to other nodes (comma-separated, e.g. \"10.0.0.0/8,192.168.0.0/24\") or empty string to not advertise routes")
 	upf.BoolVar(&upArgs.advertiseDefaultRoute, "advertise-exit-node", false, "offer to be an exit node for internet traffic for the tailnet")
+	upf.BoolVar(&upArgs.autoUpdates, "auto-updates", false, "HIDDEN: automatically update to the latest available version in the background")
```

Comment:
let's not make this available via deprecated "up"; only "set".

Also, I think it should be "update" (verb) not "updates" (noun)

Reply:
we might want more than a bool. Security only? What time schedule? What's the plan for configuring stuff like that?

Reply:
Moved to `set` and added a comment in `newUpFlagSet` for future readers to avoid adding flags there.
Also made the `ipn.Prefs` field a struct instead of bool to account for other features in the future.
I'd like to start simple (client polling daily with some jitter) and add customization later on. First goal is to get updates working in general :sweat_smile: